make/docker.mk

```diff
@@ -0,0 +1,67 @@
+#########################################
+# Building Docker Image
+#
+# This uses a multi-stage build file. The first stage is a builder (that might
+# be large in size). After the build has succeeded, the statically linked
+# binary is copied to a new image that is optimized for size.
+#########################################
+
+docker-prepare:
+	# Ensure, we can build for ARM architecture
+	[ -f /proc/sys/fs/binfmt_misc/qemu-arm ] || docker run --rm --privileged docker/binfmt:a7996909642ee92942dcd6cff44b9b95f08dad64
```

Comment:
What is `docker run --rm --privileged docker/binfmt:a7996909642ee92942dcd6cff44b9b95f08dad64`?
This will be executed if we run this on macOS.

Reply:
https://www.docker.com/blog/multi-platform-docker-builds/ -- apparently in order to build docker for multiple architectures you need to register the platforms with the kernel.